app/src/main/java/com/andela/philskiiiwalker/levelup/presenter/GithubUsersPresenter.java

```diff
@@ -4,35 +4,41 @@
 import android.support.annotation.NonNull;
 import android.support.v7.widget.RecyclerView;
 import android.util.Log;
+import android.view.View;
 
 import com.andela.philskiiiwalker.levelup.model.GithubUsers;
 import com.andela.philskiiiwalker.levelup.model.GithubUsersResponse;
 import com.andela.philskiiiwalker.levelup.adapter.GithubUsersAdapter;
 
 import com.andela.philskiiiwalker.levelup.service.GithubService;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import retrofit2.Call;
 import retrofit2.Callback;
 import retrofit2.Response;
 
 public class GithubUsersPresenter {
+    View view;
     private Context context;
 
-    public GithubUsersPresenter(Context context) {
-        this.context = context;
+    public GithubUsersPresenter(View view) {
+        this.view = view;
     }
 
-    public void getGithubUsers(final RecyclerView recyclerView) {
-        Log.d("presenter","getting users");
+    public interface View {
```

Comment:
@Kalela 
You can try extracting your interface to a separate package called `contract` and creating an interface called `MainActivityContract` that has two other interfaces`(MainView and MainPresenter - can always name it whatever you want, but be descriptive)` inside of it
Then you can add abstract methods specific to the `View` and `Presenter` respectively.
As such, the only connection between your `View` and `Presenter` will only be through the `contract(interface)`... this promotes `decoupling` and `separation of concern`

Reply:
@chykehyman I would like to ask as to why it would be better creating the `contract` package and the interfaces. are their resources that one could use to gain more knowledge in this?